compiler/src/iree/compiler/Codegen/Common/StripCompilationInfoPass.cpp

```diff
@@ -57,13 +56,33 @@ struct StripLinalgOpCompilationInfo final
   }
 };
 
+struct StripAttentionOpCompilationInfo final
+    : OpRewritePattern<IREE::LinalgExt::AttentionOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(IREE::LinalgExt::AttentionOp attentionOp,
+                                PatternRewriter &rewriter) const override {
+    if (getCompilationInfo(attentionOp)) {
+      eraseCompilationInfo(attentionOp);
+    }
+
+    if (getLoweringConfig(attentionOp)) {
+      eraseLoweringConfig(attentionOp);
+    }
+
+    if (attentionOp.getDecompositionConfigAttr()) {
+      attentionOp.removeDecompositionConfigAttr();
+    }
```

Comment:
This is wrong. DecompositionConfig is part of the operation's definition. You can only drop qk_attrs/pv_attrs in it. This can cause miscompiles.

Reply:
For example, the decomposition config can contain a "use_exp2" attribute, which specifies that the operation must use exp2 during lowering. If you use this patch, you would get numerically incorrect result.